src/libstd/io/mod.rs

```diff
@@ -606,6 +1084,22 @@ pub trait BufRead: Read {
     ///
     /// This function will yield errors whenever `read_until` would have also
     /// yielded an error.
+    ///
+    /// # Examples
+    ///
+    /// A locked standard input implements `BufRead`:
```

Comment:
Could this comment be expanded to explain what this example is doing? From a glance it's a little unclear/arbitrary as it's just splitting on `'a'` bytes